File: scripts/export-sqlite.py
#!/usr/bin/env python3
"""SQLite → sqlite-export.json (Prisma 모델명 기준 camelCase key)"""
import sqlite3, json, sys
from pathlib import Path

DB = Path("backup/phase0/dev.db.bak")
OUT = Path("sqlite-export.json")

TABLE_MAP = {
    "settings":            "settings",
    "users":               "users",
    "posts":               "posts",
    "attachments":         "attachments",
    "fee_statuses":        "feeStatuses",
    "fee_payment_accounts":"feePaymentAccounts",
    "standing_committees": "standingCommittees",
    "separate_registries": "separateRegistries",
    "resolutions":         "resolutions",
    "popups":              "popups",
    "hero_configs":        "heroConfigs",
    "board_settings":      "boardSettings",
    "rules":               "rules",
    "media_folders":       "mediaFolders",
    "file_assets":         "fileAssets",
    "comments":            "comments",
    "likes":               "likes",
}

con = sqlite3.connect(DB)
con.row_factory = sqlite3.Row
result = {}
total = 0

for table, key in TABLE_MAP.items():
    try:
        rows = [dict(r) for r in con.execute(f'SELECT * FROM "{table}"')]
        result[key] = rows
        print(f"  {key}: {len(rows)}건")
        total += len(rows)
    except sqlite3.OperationalError as e:
        print(f"  {key}: SKIP ({e})", file=sys.stderr)
        result[key] = []

con.close()
OUT.write_text(json.dumps(result, ensure_ascii=False, default=str), encoding="utf-8")
print(f"\n총 {total}건 → {OUT}")

File: scripts/export-sqlite.mjs
/**
 * SQLite → JSON export 스크립트
 * 실행: DATABASE_URL="file:./backup/phase0/dev.db.bak" node scripts/export-sqlite.mjs
 */
import { PrismaClient } from '@prisma/client';
import { writeFileSync } from 'fs';

const prisma = new PrismaClient();

async function exportData() {
    console.log('SQLite 데이터 export 시작...\n');

    const [
        settings,
        users,
        posts,
        attachments,
        feeStatuses,
        feePaymentAccounts,
        standingCommittees,
        separateRegistries,
        resolutions,
        popups,
        heroConfigs,
        boardSettings,
        rules,
        mediaFolders,
        fileAssets,
        comments,
        likes,
    ] = await Promise.all([
        prisma.settings.findMany(),
        prisma.user.findMany(),
        prisma.post.findMany(),
        prisma.attachment.findMany(),
        prisma.feeStatus.findMany(),
        prisma.feePaymentAccount.findMany(),
        prisma.standingCommittee.findMany(),
        prisma.separateRegistry.findMany(),
        prisma.resolution.findMany(),
        prisma.popup.findMany(),
        prisma.heroConfig.findMany(),
        prisma.boardSettings.findMany(),
        prisma.rule.findMany(),
        prisma.mediaFolder.findMany(),
        prisma.fileAsset.findMany(),
        prisma.comment.findMany(),
        prisma.like.findMany(),
    ]);

    const data = {
        settings, users, posts, attachments,
        feeStatuses, feePaymentAccounts, standingCommittees,
        separateRegistries, resolutions, popups, heroConfigs,
        boardSettings, rules, mediaFolders, fileAssets,
        comments, likes,
    };

    writeFileSync('sqlite-export.json', JSON.stringify(data, null, 2), 'utf-8');

    console.log('export 완료:');
    for (const [key, val] of Object.entries(data)) {
        console.log(`  ${key}: ${val.length}건`);
    }

    await prisma.$disconnect();
}

exportData().catch(e => {
    console.error('export 실패:', e);
    prisma.$disconnect();
    process.exit(1);
});
